client/actions/assignments/notifications.js

```diff
@@ -7,6 +7,20 @@ import {ASSIGNMENTS, WORKSPACE, PLANNING} from '../../constants';
 import {lockUtils, assignmentUtils, gettext, isExistingItem} from '../../utils';
 import {hideModal, showModal} from '../index';
 
+const _notifyAssignmentEdited = (assignmentId, sessionId) => (
+    (dispatch, getState, {notify}) => {
+        const currentAssignmentId = selectors.getCurrentAssignmentId(getState());
+        const currentSessionId = selectors.general.sessionId(getState());
+
+        if (assignmentId === currentAssignmentId && sessionId !== currentSessionId) {
+            notify.warning(gettext('The Assignment you were viewing was removed.'));
+            dispatch(assignments.ui.closePreview());
```

Comment:
The preview should always close if `assignmentId === currentAssignmentId`, regardless of the session.